src/compiler/c.rs

```diff
@@ -340,42 +340,51 @@ where
                     let mut updated = false;
                     let hit = preprocessor_cache_entry
                         .lookup_result_digest(preprocessor_cache_mode_config, &mut updated);
+
+                    let mut update_failed = false;
                     if updated {
                         // Time macros have been found, we need to update
                         // the preprocessor cache entry. See [`PreprocessorCacheEntry::result_matches`].
                         debug!(
                             "Preprocessor cache updated because of time macros: {preprocessor_key}"
                         );
-                        storage.put_preprocessor_cache_entry(
+
+                        if let Err(e) = storage.put_preprocessor_cache_entry(
                             preprocessor_key,
                             preprocessor_cache_entry,
-                        )?;
+                        ) {
+                            debug!("Failed to update preprocessor cache: {}", e);
+                            update_failed = true;
+                        }
```

Comment:
I'm unsure about this behavior. It's based on how `compiler.rs` handles errors.

Realistically, errors shouldn't be used here, but it seems to be how it's handled with GCS read-only, so I did it the same way. I thought about changing it, but I'd like to hear the input about someone more knowledgeable about sccache's internals first. `Storage::check()` seems to be a great candidate, although it only seems to be used for a print and is very expensive for GCS (which could be changed, of course).